src/dmd/tocsym.d

```diff
@@ -353,6 +353,24 @@ Symbol *toSymbol(Dsymbol s)
             else if (fd.isMember2() && fd.isStatic())
                 f.Fflags |= Fstatic;
 
+            if (fd.inlining == PINLINE.default_ && global.params.useInline ||
+                fd.inlining == PINLINE.always)
+            {
+                // this is copied from inline.d
+                if (!fd.fbody ||
+                    fd.ident == Id.ensure ||
+                    (fd.ident == Id.require &&
+                     fd.toParent().isFuncDeclaration() &&
+                     fd.toParent().isFuncDeclaration().needThis()) ||
+                                (fd.isSynchronized() ||
+                                 fd.isImportedSymbol() ||
+                                 fd.hasNestedFrameRefs() ||
+                                 (fd.isVirtual() && !fd.isFinalFunc())))
```

Comment:
It would be nice to have a comment saying why these can't be inlined.

Reply:
Right now I'm just trying to get it through the test suite.